src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.cs

```diff
@@ -206,6 +206,77 @@ public static partial class AsnDecoder
             return ret;
         }
 
+        /// <summary>
+        ///   Decodes the data in <paramref name="source"/> as a length value under the specified
+        ///   encoding rules.
+        /// </summary>
+        /// <param name="source">The buffer containing encoded data.</param>
+        /// <param name="ruleSet">The encoding constraints to use when interpreting the data.</param>
+        /// <param name="bytesConsumed">
+        ///   When this method returns, the number of bytes from the beginning of <paramref name="source"/>
+        ///   that contributed to the length.
+        ///   This parameter is treated as uninitialized.
+        /// </param>
+        /// <returns>
+        ///   The decoded value of the length, or <see langword="null"/> if the
+        ///   encoded length represents the indefinite length.
+        /// </returns>
+        /// <remarks>
+        ///   This method only processes the length portion of an ASN.1/BER Tag-Length-Value triplet,
+        ///   so <paramref name="source"/> needs to have already sliced off the encoded tag.
+        /// </remarks>
+        public static int? DecodeLength(
```

Comment:
This is missing `<exception>` docs. Likewise for the `Try` variant. Off the top of my head that would be `ArgumentOutOfRangeException` for a bad `ruleSet` and `AsnContentException` for "your length is bogus or invalid for the encoding rules"